### PR TITLE
#1880 Adds missing error message

### PR DESCRIPTION
**Pull request by boek**
Started Kotlinifying files as I touch them if they seem small enough. Would it be better do open those as separate PRs? Or is separate commits good?

### CODE REVIEW

**Comment by pocmo** [comment]
> Started Kotlinifying files as I touch them if they seem small enough.

That's great! 👍 

There's always a little bit of additional risk when doing that. So I wouldn't merge this today on code freeze day and wait til Monday. :)

> Would it be better do open those as separate PRs? Or is separate commits good?

Good question. I think separate commits are fine. I can still look at the actual changes in one the commits and this is good enough for me.

When "kotlinifying" files there's a high chance to cause conflicts with other PRs (In this case I'm waiting til Monday to merge another one that touches some preferences). But I guess separate PRs vs. single commit doesn't really help with that anyways?

**Comment by pocmo** [comment]
detekt is complaining about some things probably caused by the conversion (detekt only looks at Kotlin code):
https://taskcluster-artifacts.net/YMqOj8JtRLaUs6mm1qQbIA/0/public/detekt/detekt-report.html

File: app/src/main/java/org/mozilla/focus/search/ManualAddSearchEnginePreference.kt [hunk 1]
```diff
@@ -65,7 +65,8 @@ class ManualAddSearchEnginePreference(context: Context, attrs: AttributeSet) : P
 
     fun validateEngineNameAndShowError(engineName: String): Boolean {
         val errorMessage = when {
-            TextUtils.isEmpty(engineName) || !engineNameIsUnique(engineName) -> context.getString(R.string.search_add_error_empty_name)
+            TextUtils.isEmpty(engineName) -> context.getString(R.string.search_add_error_empty_name)
+            !engineNameIsUnique(engineName) -> context.getString(R.string.search_add_error_duplicate_name)
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/search/ManualAddSearchEnginePreference.kt]: We had a string for this already but didn't use it? 😮

**Comment by pocmo** [comment]
ktlint :)
> /opt/focus-android/app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt:151:17: Unexpected newline before "catch"

**Comment by boek** [comment]
@pocmo had a commit that failed to push 🤦‍♂️

File: app/src/main/java/org/mozilla/focus/search/ManualAddSearchEnginePreference.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
+package org.mozilla.focus.search
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/search/ManualAddSearchEnginePreference.kt]: nit: license header

File: app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,206 @@
+package org.mozilla.focus.settings
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt]: nit: license header

**Pull request merged**